Change MineOnlineBroadcast to use its own config file.

RhysB opened the pull request:
Change MineOnlineBroadcast to use its own config file.

RhysB committed:
Fix Plugin Name

RhysB committed:
Change Bukkit plugin to use its own config

RhysB committed:
Change HMOD to use its own config file

RhysB committed:
Discord config moved to plugin config

RhysB committed:
Load Bukkit config before Discord

RhysB committed:
Read Null strings correctly in Bukkit

craftycodie commented on src/MineOnlineConfig.java:
```diff
@@ -0,0 +1,87 @@
+        write();
+        propertiesFile.save();
+    }
+    public void write() {
```
Please keep the following options in server.properties:
- max-players
- online-mode
- whitelist

craftycodie commented on src/MineOnlineBroadcast.java:
```diff
@@ -174,47 +175,39 @@ public void enable() {
+                        if (!isPublic)
+                            return;
+
+                        String ip = mineOnlineConfig.getConfigString("serverlist-ip");
```
This should fallback to the server properties if no ip/port is provided in the config.

craftycodie commented on src/gg/codie/mineonline/plugin/bukkit/MineOnlineBroadcastPlugin.java:
```diff
@@ -155,46 +156,40 @@ public void onEnable() {
+                        if (!isPublic)
+                            return;
+
+                        String ip = mineOnlineConfig.getConfigString("serverlist-ip");
```
This should fallback to the server properties if no ip/port is provided in the config.

craftycodie commented on src/gg/codie/mineonline/plugin/bukkit/MineOnlineConfig.java:
```diff
@@ -0,0 +1,87 @@
+        //Main
+        generateConfigOption("config-version", 1);
+        //Setting
+        generateConfigOption("public", true);
```
Please keep max-players, whitelist and online-mode in the server.properties file.

RhysB committed:
Restore part of server.properties

Use server.properties for online mode, max players, and whitelist.
Fallback on server.properties for IP and Port

RhysB committed:
Don't try and use serverlist-port or serverlist-ip
